server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryModule.java

```diff
@@ -0,0 +1,67 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailrepository.cassandra;
+
+import java.util.List;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.components.CassandraTable;
+import org.apache.james.backends.cassandra.components.CassandraType;
+
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.google.common.collect.ImmutableList;
+
+public class CassandraMailRepositoryModule implements CassandraModule {
+
+    private final List<CassandraTable> tables;
+    private final List<CassandraType> types;
+
+    public CassandraMailRepositoryModule() {
+        tables = ImmutableList.of(
+            new CassandraTable(MailRepositoryTable.TABLE_NAME,
+                SchemaBuilder.createTable(MailRepositoryTable.TABLE_NAME)
+                    .ifNotExists()
+                    .addPartitionKey(MailRepositoryTable.NAME, DataType.text())
+                    .addClusteringColumn(MailRepositoryTable.CONTENT_ID, DataType.text())
+                    .addColumn(MailRepositoryTable.MESSAGE_SIZE, DataType.bigint())
+                    .addColumn(MailRepositoryTable.STATE, DataType.text())
+                    .addColumn(MailRepositoryTable.ATTRIBUTES, DataType.map(DataType.text(), DataType.text()))
+                    .addColumn(MailRepositoryTable.ERROR_MESSAGE, DataType.text())
+                    .addColumn(MailRepositoryTable.REMOTE_HOST, DataType.text())
+                    .addColumn(MailRepositoryTable.REMOTE_ADDR, DataType.text())
+                    .addColumn(MailRepositoryTable.LAST_UPDATED, DataType.date())
+                    .addColumn(MailRepositoryTable.PER_RECIPIENT_SPECIFIC_HEADERS, DataType.map(DataType.text(), DataType.text()))
```

Comment:
It's more Map(text(), pair(text(), text()); as headers are composed of field and value here.

Reply:
BTW maybe we could have a Header type...